crates/cext/src/transpiler/passes/remove_identity_equiv.rs

```diff
@@ -0,0 +1,101 @@
+///
+/// $bar{F} = \frac{1 + d F_{\text{process}}}{1 + d},F_{\text{process}} =
+/// \frac{|\mathrm{Tr}(G)|^2}{d^2}$
+///
+/// where $d = 2^n$ is the dimension of the gate for $n$ qubits.
+///
+/// @param circuit A pointer to the circuit to run RemoveIdentityEquivalent on
+/// @param target The target for the RemoveIdentityEquivalent pass. If ``approximation_degree`` is set to
+/// ``NAN`` the tolerance for determining whether an operation is equivalent to
+/// identity will be set to the reported error rate in the target. Otherwise
+/// the ``target`` is not used as the tolerance is independent of the target.
+/// @param approximation_degree The degree to approximate for the equivalence check. This can be a
+/// floating point value between 0 and 1, or ``NAN``. If the value is 1 this does not
+/// approximate above the floating point precision. For a value < 1 this is used as a
+/// scaling factor for the cutoff fidelity. If the value is ``NAN`` this approximates up
+/// to the fidelity for the gate specified in ``target``.
+///
+/// @return QkCircuit A pointer to a new circuit which results from running the pass.
+///
+/// # Example
+///
+/// ```c
+///     QkTarget *target = qk_target_new(5)
+///     QkTargetEntry *cx_entry = qk_target_entry_new(QkGate_CX);
+///     for (uint32_t i = 0; i < current_num_qubits - 1; i++) {
+///         uint32_t qargs[2] = {i, i + 1};
+///         double inst_error = 0.0090393 * (current_num_qubits - i);
+///         double inst_duration = 0.020039;
+///         qk_target_entry_add_property(cx_entry, qargs, 2, inst_duration, inst_error);
+///     }
+///     QkExitCode result_cx = qk_target_add_instruction(target, cx_entry);
+///     QkCircuit *qc = qk_circuit_new(4, 0);
+///     for (uint32_t i = 0; i < qk_circuit_num_qubits(qc) - 1; i++) {
+///         uint32_t qargs[2] = {i, i + 1};
+///         for (uint32_t j = 0; j<i+1; j++) {
+///             qk_circuit_gate(qc, QkGate_CX, qargs, NULL);
+///         }
+///     }
+///     qk_circuit_gate(qc, QkGate_RZ, {1,}, {0.,});
+///     QkCircuit *output_circuit = qk_transpiler_pass_standalone_remove_identity_equivalent(qc, target, 1.0);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``circuit`` or ``target`` is not a valid, non-null pointer to a ``QkCircuit`` and ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_transpiler_pass_standalone_remove_identity_equivalent(
+    circuit: *const CircuitData,
+    target: *const Target,
+    approximation_degree: f64,
+) -> *mut CircuitData {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let circuit = unsafe { const_ptr_as_ref(circuit) };
+    let target = unsafe { const_ptr_as_ref(target) };
+    let mut dag = match DAGCircuit::from_circuit_data(circuit, false, None, None, None, None) {
+        Ok(dag) => dag,
+        Err(e) => panic!("{}", e),
+    };
+    let approximation_degree = if approximation_degree.is_nan() {
```

Comment:
It seems a bit weird to only do NAN checks, but no other validation (i.e. that this value is in [0, 1]). Would it make sense to add this, or just remove all checks?

Reply:
I added a bounds check on the value if it's not NAN in: https://github.com/Qiskit/qiskit/pull/14717/commits/30b5c8f65756b7fbbf242af4622b7bf6d4cbfe4a

Reply:
That is interesting Rust syntax, I didn't know this existed!